### PR TITLE
citra  2020-12-07 -> 2021-06-22

### PR DESCRIPTION
**Pull request by bezmuth**
###### Motivation for this change
Bump citra to the latest commit.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/emulators/citra/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
 
 mkDerivation {
   pname = "citra";
-  version = "2020-12-07";
+  version = "2021-06-22";
 
   # Submodules
   src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/citra/default.nix:8]: Can you convert this to fetchFromGitHub?

Comment (bezmuth) [reply pkgs/misc/emulators/citra/default.nix:8]: I tried this, but the citra repo fetches submodules from other locations, and I'm not 100% sure how to get fetchFromGitHub to fetch submodules.

Comment (SuperSandro2000) [reply pkgs/misc/emulators/citra/default.nix:8]: https://github.com/NixOS/nixpkgs/blob/6ecbfcdb36d5b6b1daee265d3e2efad0be8ce21f/doc/builders/fetchers.chapter.md#L33

Comment (bezmuth) [reply pkgs/misc/emulators/citra/default.nix:8]: Awesome, sorry about that I couldn't find the fetchers docs.

File: pkgs/misc/emulators/citra/default.nix [hunk 2]
```diff
@@ -1,14 +1,16 @@
-{ mkDerivation, lib, fetchgit, cmake, SDL2, qtbase, qtmultimedia, boost }:
+{ mkDerivation, lib, fetchgit, cmake, SDL2, qtbase, qtmultimedia, boost, fetchFromGitHub  }:
 
 mkDerivation {
   pname = "citra";
-  version = "2020-12-07";
+  version = "2021-07-01";
 
   # Submodules
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/citra/default.nix:7]:
```suggestion
```

File: pkgs/misc/emulators/citra/default.nix [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
-  src = fetchgit {
-    url = "https://github.com/citra-emu/citra";
-    rev = "3f13e1cc2419fac837952c44d7be9db78b054a2f";
-    sha256 = "1bbg8cwrgncmcavqpj3yp4dbfkip1i491krp6dcpgvsd5yfr7f0v";
+  src = fetchFromGitHub {
+    owner = "citra-emu";
+    repo = "citra";
+    rev = "afed4953bccfb9b39558cf196492c2558ca1f109";
+    sha256 = "11sicbgxa54wwj3jrklprqiznpxjqmxrymz23lvccyybq3pnmfqj";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/citra/default.nix:13]: I don't like that we vendor so many projects.

File: pkgs/misc/emulators/citra/default.nix [hunk 4]
```diff
@@ -1,14 +1,15 @@
-{ mkDerivation, lib, fetchgit, cmake, SDL2, qtbase, qtmultimedia, boost }:
+{ mkDerivation, lib, cmake, SDL2, qtbase, qtmultimedia, boost, fetchFromGitHub  }:
 
 mkDerivation {
   pname = "citra";
-  version = "2020-12-07";
+  version = "2021-07-01";
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/citra/default.nix:5]:
```suggestion
  version = "unstable-2021-07-01";
```

File: pkgs/misc/emulators/citra/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/misc/emulators/citra/default.nix:12]: we have zstd, fmt, libusb and probably more in nixpkgs

Comment (bezmuth) [reply pkgs/misc/emulators/citra/default.nix:12]: Sorry I don't really understand what you're asking me to do here.

Comment (SuperSandro2000) [reply pkgs/misc/emulators/citra/default.nix:12]: Ideally we don't want to fetchSubmodules at all. If that is not possible than we should remove all projects which we have in nixpkgs via rm in postPatch and provide our versions of those libraries.

**Comment by JoshuaFern** [comment]
If audio is still broken unless sdl2 is selected, I believe it's because alsa-lib is a missing dependency.

https://github.com/NixOS/nixpkgs/pull/83193#issue-586039373
https://github.com/NixOS/nixpkgs/pull/84117#issuecomment-618080004

**Pull request closed without merging**